### PR TITLE
Add custom policy to lambda role

### PR DESCRIPTION
**Pull request by jamengual**
## what
* Add user provides policy to be attached to lambda role

## why
* Custom policies are needed for lambda to use resources like KMS, Secrets and others.

### CODE REVIEW

File: iam-role.tf [hunk 1]
```diff
@@ -72,3 +72,10 @@ resource "aws_iam_role_policy_attachment" "ssm" {
   policy_arn = aws_iam_policy.ssm[count.index].arn
   role       = aws_iam_role.this[0].name
 }
+
+resource "aws_iam_role_policy_attachment" "custom" {
+  count = try((local.enabled && var.custom_iam_policy_arn != null), false) ? 1 : 0
+
+  policy_arn = var.custom_iam_policy_arn
```
Comment (nitrocode) [inline iam-role.tf:79]: what if you wanted to attach more than one custom policy?

Comment (jamengual) [reply iam-role.tf:79]: good point, changed to list?

File: iam-role.tf [hunk 2]
```diff
@@ -72,3 +72,9 @@ resource "aws_iam_role_policy_attachment" "ssm" {
   policy_arn = aws_iam_policy.ssm[count.index].arn
   role       = aws_iam_role.this[0].name
 }
+
+resource "aws_iam_role_policy_attachment" "custom" {
+  for_each   = local.enabled && var.custom_iam_policy_arn != [] ? var.custom_iam_policy_arn : []
```
Comment (nitrocode) [inline iam-role.tf:77]:
```suggestion
  for_each   = local.enabled && length(var.custom_iam_policy_arns) > 0 ? var.custom_iam_policy_arns : toset([])
```

File: variables.tf [hunk 3]
```diff
@@ -239,3 +239,10 @@ variable "vpc_config" {
     subnet_ids         = list(string)
   })
 }
+
+variable "custom_iam_policy_arn" {
+  type        = set(string)
+  description = "List of custom policy to be attached to the lambda role"
+  default     = []
+
+}
```
Comment (nitrocode) [inline variables.tf:248]:
```suggestion
variable "custom_iam_policy_arns" {
  type        = set(string)
  description = "List of custom policy to be attached to the lambda role"
  default     = []
}
```

**Review by nitrocode — CHANGES_REQUESTED** [review]
Please see comments

File: iam-role.tf (hunk 2 above)
Comment (aknysh) [reply iam-role.tf]: yep, `var.custom_iam_policy_arn != []` is not a TF expression

File: variables.tf [hunk 4]
```diff
@@ -239,3 +239,9 @@ variable "vpc_config" {
     subnet_ids         = list(string)
   })
 }
+
+variable "custom_iam_policy_arns" {
+  type        = set(string)
+  description = "List of custom policy to be attached to the lambda role"
```
Comment (aknysh) [inline variables.tf:245]:
```suggestion
  description = "List of custom policies to be attached to the lambda role"
```

**Review by aknysh — DISMISSED** [review]

**Comment by aknysh** [comment]
@jamengual approved with one nitpick

**Review by nitrocode — DISMISSED** [review]

**Review by nitrocode — DISMISSED** [review]

**Review by nitrocode — APPROVED** [review]

**Pull request merged**